src/compiler/transformers/map-imports-to-path-aliases.ts

```diff
@@ -11,15 +11,28 @@ import type * as d from '../../declarations';
  * their compiler ({@link https://github.com/microsoft/TypeScript/issues/10866})
  *
  * @param config The Stencil configuration object.
+ * @param destinationFilePath The location on disk the file will be written to.
+ * @param destinationDirectory The destination on disk of the collection directory.
  * @returns A factory for creating a {@link ts.Transformer}.
  */
-export const mapImportsToPathAliases = (config: d.Config): ts.TransformerFactory<ts.SourceFile> => {
+export const mapImportsToPathAliases = (
+  config: d.ValidatedConfig,
+  destinationFilePath: string,
+  destinationDirectory: string
```

Comment:
Can we add a few test cases for these, where
- `destinationFilePath` is a parent of `destinationDirectory` and both are non-empty strings?
- `destinationDirectory` is a parent of `destinationFilePath` and both are non-empty strings?
- both are the same directory and both are non-empty strings?

this might get a little funky, since we'll probably have to rely on `normalize` in our tests for Win/Unix compatability